apps/native/app/src/screens/document-detail/document-detail.tsx

```diff
@@ -220,24 +246,81 @@ export const DocumentDetailScreen: NavigationFunctionComponent<{
     returnPartialData: true,
   })
 
+  // We want to make sure we don't include the document content if isUrgent is undefined/null since then we don't have
+  // the info from the server and don't want to make any assumptions about it just yet
+  const shouldIncludeDocument = !(
+    isUrgent === true ||
+    isUrgent === undefined ||
+    isUrgent === null
+  )
+
   // Fetch the document to get the content information
   const docRes = useGetDocumentQuery({
     variables: {
       input: {
         id: docId,
+        // If the document is urgent we need to check if the user needs to confirm reception of it before fetching the document data
+        includeDocument: shouldIncludeDocument,
       },
+      locale: locale === 'is-IS' ? 'is' : 'en',
     },
     fetchPolicy: 'no-cache',
+    onCompleted: async (data) => {
+      const confirmation = data.documentV2?.confirmation
+      if (confirmation && !refetching && !showConfirmedAlert) {
+        RNAlert.alert(confirmation.title ?? '', confirmation.data ?? '', [
```

Comment:
I actually thought about it. But I'm not sure, since this is coming from the api and is regarding some sensitive stuff  I'm not sure we would want to add some defaults, but I will ask Dísa from Hugsmiðjan just in case 🙏